stepper/stepper.c

```diff
@@ -0,0 +1,129 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "stepper.h"
+#include "../opto-fork/opto.h"
+
+#define A_GPIO 2
+#define B_GPIO 3
+#define C_GPIO 6
+#define D_GPIO 13
+#define COIL_COUNT 4
+
+void init_stepper() {
+    static const int stepper_pin[COIL_COUNT] = {A_GPIO,
+                                                B_GPIO,
+                                                C_GPIO,
+                                                D_GPIO};
+
+    for (int stepper_i = 0; stepper_i < COIL_COUNT; stepper_i++) {
+        gpio_init(stepper_pin[stepper_i]);
+        gpio_set_dir(stepper_pin[stepper_i], GPIO_OUT);
+    }
+}
+
+enum step_stage {
+    A  = 0b0001,
+    AB = 0b0011,
+    B  = 0b0010,
+    BC = 0b0110,
+    C  = 0b0100,
+    CD = 0b1100,
+    D  = 0b1000,
+    DA = 0b1001
+};
+
+#define STEP_STATES 8
+
+void step(bool clockwise) {
+    static const uint8_t step_masks[STEP_STATES] =
+            {A, AB, B, BC, C, CD, D, DA};
```

Comment:
I made it an enum, because I'm using them also in the struct declarations below.
We could, as an alternative, define the step_masks directly with the binary values (which is slightly less clear to read),
and the struct values as well. Then we wouldn't need the enum.